File: backend/internal/cache/keys.go
package cache

import "fmt"

// TTL default (detik) — selaras ARSITEKTUR.md §9
const (
	TTLDosenMahasiswaList = 120
	TTLKrsPending         = 60
	TTLKrsMahasiswaList   = 120
)

// KRS / antrian persetujuan
func KeyKrsPendingDosen(dosenID string) string {
	return fmt.Sprintf("krs:pending:dosen:%s", dosenID)
}

func KeyKrsPendingStaff() string {
	return "krs:pending:staff"
}

func KeyKrsMahasiswaList(mahasiswaID string) string {
	return fmt.Sprintf("krs:mhs:%s:list", mahasiswaID)
}

// Master data (invalidate on CRUD / PATCH DPA — Slice 4)
func KeyDosenList() string {
	return "dosen:list"
}

func KeyDosenDetail(dosenID string) string {
	return fmt.Sprintf("dosen:%s", dosenID)
}

func KeyMahasiswaList() string {
	return "mahasiswa:list"
}

func KeyMahasiswaDetail(mahasiswaID string) string {
	return fmt.Sprintf("mahasiswa:%s", mahasiswaID)
}

File: backend/src/shared/cache/keys.ts
export const TTL_DOSEN_MAHASISWA_LIST = 120;
export const TTL_KRS_PENDING = 60;
export const TTL_KRS_MAHASISWA_LIST = 120;

export function keyKrsPendingDosen(dosenId: string): string {
  return `krs:pending:dosen:${dosenId}`;
}

export function keyKrsPendingStaff(): string {
  return 'krs:pending:staff';
}

export function keyKrsMahasiswaList(mahasiswaId: string): string {
  return `krs:mhs:${mahasiswaId}:list`;
}

export function keyDosenList(): string {
  return 'dosen:list';
}

export function keyDosenDetail(dosenId: string): string {
  return `dosen:${dosenId}`;
}

export function keyMahasiswaList(): string {
  return 'mahasiswa:list';
}

export function keyMahasiswaDetail(mahasiswaId: string): string {
  return `mahasiswa:${mahasiswaId}`;
}
